### DISCUSSION TITLE
Trying out stable diffusion for the first time. Some feedback would be great

### DISCUSSION THREAD

**Discussion by demon8555** [lshqqytiger/stable-diffusion-webui-directml#384] [General]
I'm writing this so it could help me and other beginners. A couple of suggestions and feedback would be nice. Here's what I've done so far in stable diffusion in the last 7 days:
I'm on Win10, 32GB RAM, RX 6800 with 16GB VRAM, latest video driver. I started with stable diffusion a week ago roughly. I first tried fooocus, and within a couple of hours I got it working, and was able to generate quality pictures.
But the speed wasn't so good, like 6s/it (yeah not it/s), or even worse. I heard that with a1111, you could get even up 20 it/s (!), which would be a huge difference on AMD.
I tried to get this project up and running, but struggled with it for a couple of days. Probably the timing wasn't so good, because all of a sudden the guides and youtube videos got out of date. Things changed (like don't use the --onnx parameter) and other stuffs (and fresh bugs too).
But after a couple of days webui finally up and running, and was able to generate my first picture. I even managed to run the juggernautXL_v8Rundiffusion model which is currently the default in fooocus. And the speed is actually double, around 3s/it (1024x1024). With a SD 1.5 model, and only 512x512, I get around 3-5 it/s. Which is nice, but it's not even using refiner, hires, whatever. So I'm not sure how to achieve those 20 it/s speeds.

I'm currenlty here, and having some issues which I can't get my head wrapped around:
1. Image quality: I tried the same prompt with the same model (juggernautXL_v8Rundiffusion) on both fooocus and a1111. 1024x1024, 30 steps both. This is fooocus:
![2024-02-13_11-58-54_2277](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/159694052/4be52d8c-4fa4-4bb9-8165-436eef16a852)
And this is a1111. it's clearly missing something, because quality is a lot worse:
![00004](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/159694052/bb042a58-a88a-43c9-924e-619c1d5cb36b)
But what is it?

3. Image consistency: getting perfectly the same image in fooocus is easy, use the same settings (same seed), and you get the same picture, which I like. Somehow in a1111 this doesn't work, I always get a random image (yes same seed and everything). Any suggestions on that? This only me or is it normal here?
4. hires fix doesn't seems to work at all. Not sure if this is a new bug or what, no matter how I setup hires fix, it just doesn't increase the image size at all. no change in the console too, same iteration speed, no trace that it's doing anything.
5. Stability: currently with the only sdxl model I tried (juggernautXL_v8Rundiffusion) I'm able to generate only once, then I get "The GPU will not respond to more commands", and I have to relaunch the whole app. SD 1.5 seems fine atm.
6. Speed: it's faster already, although there are issues, so this speed my change later downward.. Any feedback would be nice, what others can achieve with similar conditions maybe

I'm not sure what belongs to this repository, and what would be better in the a1111 repo. I believe as a win + AMD user, this is the only option that works (right?), hence I'm posting here.

So some feedback would nice, and probably other beginners can learn from it too! Thanks in advance!

**Comment by songib** [comment]
go here, it discusses a bunch of settings. https://github.com/lshqqytiger/stable-diffusion-webui-directml/discussions/84

**Comment by demon8555** [comment]
If anyone still struggling out there: I switched to SD.Next with zluda. All of the issues above solved with that, and it's much faster.

**Reply by demon8555** [reply]
Actually switched to the original a1111 webui repo, got zluda working with that one too. It's working great.. speed is even a little bit faster than sd next with zluda. the ui is much better.. (I didn't liked sd next ui at all). And all the tutorials are for a1111...

**Reply by pw405** [reply]
oh I didn't know this was a thing... is there a guide?

**Reply by demon8555** [reply]
Combine the base guide on SD.Next:
https://github.com/vladmandic/automatic/wiki/ZLUDA
with some additional things in this discussion on automatic1111:
https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/14946
And it should work eventually..
